### DISCUSSION TITLE
Golden ratio auto-resize windows

### DISCUSSION THREAD

**Discussion by stevenbarragan** [helix-editor/helix#10276] [Ideas]
form https://github.com/roman/golden-ratio

Resize automatically the windows you are working on to the size specified in the "Golden Ratio". The window that has the main focus will have the perfect size for editing, while the ones that are not being actively edited will be re-sized to a smaller size that doesn't get in the way, but at the same time will be readable enough to know it's content.

I love this feature I hope it can be ported over helix.

**Comment by David-Else** [comment]
I think automatically resizing the current window to be the best size for editing is a cool idea, but surely columns/characters is better than ratio?

50 - 75 characters is optimum for readability according to https://baymard.com/blog/line-length-readability.  I think the programming community has settled on 80 chars, which works very well. When we get soft word wrapping then some kind of 'zen mode' like https://github.com/folke/zen-mode.nvim would be amazing.

**Comment by erasin** [comment]
@David-Else I thought you might be interested. [#2704 zen mode like](https://github.com/helix-editor/helix/discussions/6468)

**Comment by David-Else** [comment]
@erasin Thanks! I will check that out!

Did you know about this 'zen mode' pull request? https://github.com/helix-editor/helix/pull/3239 . It was recently merged with master and had a review from [archseer](https://github.com/archseer), but the author has no more time to finish it, I am learning Rust but not really up to taking it over yet, what do you make of it?

**Comment by erasin** [comment]
@David-Else I don't know enough about the helix code，has still need to read the code.
